File: SmartButler/app/src/main/java/net/away0x/italker/smartbutler/utils/ShareUtils.java
package net.away0x.italker.smartbutler.utils;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * SharedPreferences 封装
 */
public class ShareUtils {
    public static final String NAME = "config";

    // 获取和设置
    public static void putString(Context context, String key, String value) {
        SharedPreferences sp = context.getSharedPreferences(NAME, Context.MODE_PRIVATE);
        sp.edit().putString(key, value).commit();
    }

    public static String getString(Context context, String key, String defaultValue) {
        SharedPreferences sp = context.getSharedPreferences(NAME, Context.MODE_PRIVATE);
        return sp.getString(key, defaultValue); // defaultValue 为默认值
    }

    public static void putInt(Context context, String key, int value) {
        SharedPreferences sp = context.getSharedPreferences(NAME, Context.MODE_PRIVATE);
        sp.edit().putInt(key, value).commit();
    }

    public static int getInt(Context context, String key, int defaultValue) {
        SharedPreferences sp = context.getSharedPreferences(NAME, Context.MODE_PRIVATE);
        return sp.getInt(key, defaultValue);
    }

    public static void putBoolean(Context context, String key, boolean value) {
        SharedPreferences sp = context.getSharedPreferences(NAME, Context.MODE_PRIVATE);
        sp.edit().putBoolean(key, value).commit();
    }

    public static boolean getBoolean(Context context, String key, boolean defaultValue) {
        SharedPreferences sp = context.getSharedPreferences(NAME, Context.MODE_PRIVATE);
        return sp.getBoolean(key, defaultValue);
    }

    // 删除单个
    public static void deleteSingle(Context context, String key) {
        SharedPreferences sp = context.getSharedPreferences(NAME, Context.MODE_PRIVATE);
        sp.edit().remove(key).commit();
    }

    // 删除全部
    public static void deleteAll(Context context) {
        SharedPreferences sp = context.getSharedPreferences(NAME, Context.MODE_PRIVATE);
        sp.edit().clear().commit();
    }
}

File: smartButler/app/src/main/java/com/away0x/smartbutler/utils/ShareUtils.kt
package com.away0x.smartbutler.utils

import android.content.Context
import android.content.SharedPreferences

object ShareUtils {

    const val NAME = "config"

    private fun getSp(context: Context): SharedPreferences {
        return context.getSharedPreferences(NAME, Context.MODE_PRIVATE)
    }

    fun putString(context: Context, key: String, value: String) {
        getSp(context).edit().putString(key, value).apply()
    }

    fun getString(context: Context, key: String, defaultValue: String): String {
        return getSp(context).getString(key, defaultValue) ?: ""
    }

    fun putInt(context: Context, key: String, value: Int) {
        getSp(context).edit().putInt(key, value).apply()
    }

    fun getInt(context: Context, key: String, defaultValue: Int): Int {
        return getSp(context).getInt(key, defaultValue)
    }

    fun putBoolean(context: Context, key: String, value: Boolean) {
        getSp(context).edit().putBoolean(key, value).apply()
    }

    fun getBoolean(context: Context, key: String, defaultValue: Boolean): Boolean {
        return getSp(context).getBoolean(key, defaultValue)
    }

    fun remove(context: Context, key: String) {
        getSp(context).edit().remove(key).apply()
    }

    fun clear(context: Context) {
        getSp(context).edit().clear().apply()
    }

}